### PR TITLE
Add export to JSON option

### PR DESCRIPTION
**Pull request by RCoeurjoly**
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/6065)

### CODE REVIEW

File: plugins/evernote.koplugin/main.lua [hunk 1]
```diff
@@ -17,6 +17,7 @@ local N_ = _.ngettext
 local slt2 = require('slt2')
 local MyClipping = require("clip")
 local realpath = require("ffi/util").realpath
+local json = require("json")
```
Comment (Frenzie) [inline plugins/evernote.koplugin/main.lua:20]: Nit about order (see coding style). Clearly this plugin was never updated with them though so not a big deal for this PR.

File: plugins/evernote.koplugin/main.lua [hunk 2]
```diff
@@ -44,14 +46,19 @@ function EvernoteExporter:init()
...
     self.txt_export = settings.txt_export or false
-    --- @todo Is this if block necessarry? Nowhere in the code they are assigned both true.
+    self.json_export = settings.json_export or false
+    --- @todo Is this if block necessary? Nowhere in the code they are assigned both true.
     -- Do they check against external modifications to settings file?
 
     if self.html_export then
-        self.txt_export = false
-        self.joplin_export = false
+      self.txt_export = false
```
Comment (Frenzie) [inline plugins/evernote.koplugin/main.lua:54]: Could you please stick to the four-space indentation?

File: plugins/evernote.koplugin/main.lua [hunk 3]
```diff
@@ -233,16 +242,16 @@ function EvernoteExporter:addToMainMenu(menu_items)
                         keep_menu_open = true,
                         callback = function()
                             UIManager:show(InfoMessage:new{
-                                text = T(_([[You can enter your auth token on your computer by saving an empty token. Then quit KOReader, edit the evernote.joplin_token field in %1/settings.reader.lua after creating a backup, and restart KOReader once you're done.
+                                               text = T(_([[You can enter your auth token on your computer by saving an empty token. Then quit KOReader, edit the evernote.joplin_token field in %1/settings.reader.lua after creating a backup, and restart KOReader once you're done.
```
Comment (Frenzie) [inline plugins/evernote.koplugin/main.lua:245]: Like this one, it's very weird.

File: plugins/evernote.koplugin/main.lua [hunk 4]
```diff
@@ -122,7 +130,7 @@ function EvernoteExporter:addToMainMenu(menu_items)
                                 self:login()
                             end
                         }
-                    } or nil
+                                                       } or nil
```
Comment (Frenzie) [inline plugins/evernote.koplugin/main.lua:133]: Did you use some automated method for this? There are a few really weird changes here, like this line and on lines further down below. I really just meant in the previous diff.

File: plugins/evernote.koplugin/main.lua [hunk 5]
```diff
@@ -233,16 +242,16 @@ function EvernoteExporter:addToMainMenu(menu_items)
                         keep_menu_open = true,
                         callback = function()
                             UIManager:show(InfoMessage:new{
-                                text = T(_([[You can enter your auth token on your computer by saving an empty token. Then quit KOReader, edit the evernote.joplin_token field in %1/settings.reader.lua after creating a backup, and restart KOReader once you're done.
+                                               text = T(_([[You can enter your auth token on your computer by saving an empty token. Then quit KOReader, edit the evernote.joplin_token field in %1/settings.reader.lua after creating a backup, and restart KOReader once you're done.
 
-To export to Joplin, you must forward the IP and port used by this plugin to the localhost:port on which Joplin is listening. This can be done with socat or a similar program. For example:
+                                                   To export to Joplin, you must forward the IP and port used by this plugin to the localhost:port on which Joplin is listening. This can be done with socat or a similar program. For example:
```
Comment (Frenzie) [inline plugins/evernote.koplugin/main.lua:247]: And this is completely wrong. ;-)

Comment (RCoeurjoly) [reply plugins/evernote.koplugin/main.lua:247]: Let me close this pull this pull request, fix the things and create another one.
It's my first time with Lua and yes, that was emacs indenting. :)

**Pull request closed without merging**